### PR TITLE
Swatch fix

### PR DESCRIPTION
**Pull request by samanpwbb**
Fixes https://github.com/mapbox/tm2/issues/274 / fixes https://github.com/mapbox/tm2/issues/273 properly / moves swatches to front of color: 
screen shot 2014-05-26 at 2 28 55 am

Not ready yet. Notes inline.

### CODE REVIEW

File: app/codemirror.tm2.palette.js [hunk 1]
```diff
@@ -22,55 +22,32 @@
...
+            doc.findMarks({
+                line:doc.getLineNumber(line),
+                ch:0
+            }, {
+                line:doc.getLineNumber(line) + 1,
+                ch:0
+            }).filter(isPaletteMark).forEach(clear);
+
+
             while ((match = text.match(COLOR_PATTERN))) {
```
Comment (samanpwbb) [inline app/codemirror.tm2.palette.js:42]: Each match is returned as it's own object even though what I'd like to get back is an array including every match in a line. This seems wrong. Even worse, every match on a line has the same 'index' value as the first match, so it's hard to have control over clearing and placing swatches. I'm sure there's a better way to implement this but I could use some pointers before continuing on my own.

Comment (samanpwbb) [inline app/codemirror.tm2.palette.js:42]: This wasn't a real problem. Mostly I wasn't understanding parts of the code (not familiar with while loops when I wrote this - enough messing around and now I understand). I did end up fixing the problem with multiple swatches on the same line.

**Comment by samanpwbb** [comment]
One issue left: copy + paste only creates swatches for the first line, ignores the rest.

**Comment by samanpwbb** [comment]
Just fixed that last issue. This is ready to go.

@yhahn can you give this a really quick review just to make sure I'm not doing anything horrible.

**Comment by yhahn** [comment]
Looks great.

**Pull request merged**